internal/adapters/events/events_forwarder.go

```diff
@@ -68,19 +70,19 @@ func (f *eventsForwarder) ForwardRoomEvent(ctx context.Context, eventAttributes
 	case events.Status:
 		event, err := f.buildRoomStatusMessage(eventAttributes, forwarder)
 		if err != nil {
-			return errors.NewErrInvalidArgument("failed to build room status message: %s", err)
+			return codes.InvalidArgument, errors.NewErrInvalidArgument("failed to build room status message: %s", err)
 		}
 
 		eventResponse, err := f.forwarderClient.SendRoomStatus(ctx, forwarder, &event)
 
 		return handlerGrpcClientResponse(forwarder, eventResponse, err)
 	}
 
-	return errors.NewErrUnexpected("failed to forward event room. event type doesn't exists \"%s\"", eventAttributes.EventType)
+	return codes.NotFound, errors.NewErrUnexpected("failed to forward event room. event type doesn't exists \"%s\"", eventAttributes.EventType)
```

Comment:
I think we should return `codes.Unknown` and leave `NotFound` strictly to 404 of the client